cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/PythonAddDeclarationsPass.kt

```diff
@@ -79,59 +78,105 @@ class PythonAddDeclarationsPass(ctx: TranslationContext) : ComponentPass(ctx) {
         }
     }
 
-    /*
-     * Return null when not creating a new decl
+    /**
+     * This function will create a new dynamic [VariableDeclaration] if there is a write access to
```

Comment:
This function does not check whether there is a write access. However, the doc implies this. Maybe rephrase to "should be called when handling write access invocations"?

Reply:
I was actually thinking in adding another check so I can do that, should be minimal effort for the compiler.

Reply:
Then I can also get rid of the weird check if this is part of a call expression

Reply:
There is a bug with write references in a named expression. https://github.com/Fraunhofer-AISEC/cpg/pull/1785 needs to be resolved first.

Reply:
Check implemented